### PR TITLE
Initial release of adventure agency

### PR DESCRIPTION
**Pull request by Lemongrass3110**
* **Addressed Issue(s)**: None

* **Server Mode**: Both

* **Description of Pull Request**: 
Adds support for the adventure agency feature.
Some additional cleanups in web-server.

### CODE REVIEW

**Comment by JohnnyPlayy** [comment]
What beautiful codes! Another incredible job. 😻

File: sql-files/web.sql [hunk 1]
```diff
@@ -48,3 +48,24 @@ CREATE TABLE IF NOT EXISTS `merchant_configs` (
   `data` longtext NOT NULL,
   PRIMARY KEY (`world_name`, `account_id`, `char_id`)
 ) ENGINE=MyISAM;
+
+--
+-- Table structure for table `party_bookings`
+--
+
+CREATE TABLE IF NOT EXISTS `party_bookings` (
```
Comment (vstumpf) [inline sql-files/web.sql:56]: if this is handled by the map server, shouldn't it go in main.sql?

File: src/web/partybooking_controller.cpp [hunk 2]
```diff
@@ -0,0 +1,530 @@
...
+	uint16 maximum_level;
+	std::string comment;
+
+public:
+	std::string to_json( std::string& world_name );
+};
+
+std::string s_party_booking_entry::to_json( std::string& world_name ){
+	return
+		"{ \"AID\": " + std::to_string( this->account_id ) +
```
Comment (vstumpf) [inline src/web/partybooking_controller.cpp:50]: soon i'll add nlohmann and we can get rid of this :D

**Comment by gravity-ro** [comment]
Bug report: 
When join the party disconnect from server
Error in Map Server

[Warning]: clif_parse: Received packet 0x0ae6 with expected packet length 12, but only 10 bytes remaining, disconnecting session #3.

**Comment by vstumpf** [comment]
> Error in Map Server
> 
> [Warning]: clif_parse: Received packet 0x0ae6 with expected packet length 12, but only 10 bytes remaining, disconnecting session`#3`.

Can you be more specific, what packetver are you using?

**Comment by ecdarreola** [comment]
Hmmm can't seem to find the button to register in party booking using 2020-03-04RagexeRE. See attached.
image

And I don't see any request going to web-server after using the PR. See below.
image

Am I missing something?

**Comment by secretdataz** [comment]
@ecdarreola You client doesn't even try to call the party booking api. Likely a lua file issue.

**Comment by ecdarreola** [comment]
> @ecdarreola Your client doesn't even try to call the party booking api. Likely a lua file issue.

The lua files in service_korea folder that has the externalsettings has been updated to use the IP of web-server which is localhost. Are there any lua files needs to be updated for party booking?

**Comment by Lemongrass3110** [comment]
> > @ecdarreola Your client doesn't even try to call the party booking api. Likely a lua file issue.
> 
> The lua files in service_korea folder that has the externalsettings has been updated to use the IP of web-server which is localhost. Are there any lua files needs to be updated for party booking?

Make sure that you configure the right property. I think in the old clients the property for the URL was called different and for party booking they added a new property: `AssistAddr = "127.0.0.1:8888"`

**Comment by Lemongrass3110** [comment]
> Bug report: When join the party disconnect from server Error in Map Server
> 
> [Warning]: clif_parse: Received packet 0x0ae6 with expected packet length 12, but only 10 bytes remaining, disconnecting session #3.

@gravity-ro oh. you are back? Since you AGAIN were unable to apply the PR correctly (0xae6 is added in this PR) I am thinking of giving you another thinking time before you can spam comments all over our repo again. :) So maybe think twice before commenting and check if you did everything correctly.

**Comment by ecdarreola** [comment]
> > > @ecdarreola Your client doesn't even try to call the party booking api. Likely a lua file issue.
> > 
> > 
> > The lua files in service_korea folder that has the externalsettings has been updated to use the IP of web-server which is localhost. Are there any lua files needs to be updated for party booking?
> 
> Make sure that you configure the right property. I think in the old clients the property for the URL was called different and for party booking they added a new property: `AssistAddr = "127.0.0.1:8888"` But it is as @secretdataz said, there are no requests to your web-server and that is why you cannot see the button to register a booking.

Yeah I noticed that the new client uses AssistAddr. I will try to find old reference of lua for externalsettings for 2020-03-04 client first. Thanks for the info.

**Comment by gravity-ro** [comment]
> > Bug report: When join the party disconnect from server Error in Map Server
> > [Warning]: clif_parse: Received packet 0x0ae6 with expected packet length 12, but only 10 bytes remaining, disconnecting session #3.
> 
> @gravity-ro oh. you are back? Since you AGAIN were unable to apply the PR correctly (0xae6 is added in this PR) I am thinking of giving you another thinking time before you can spam comments all over our repo again. :) So maybe think twice before commenting and check if you did everything correctly.

==============================================
In fact, every script I've followed every line There might be some mistakes. But why do you think I'm spam?

If I find anything wrong, I'll report it. Have good intentions every time why is there a problem

Example of a person who reported that the button was not found. join the party He was wrong to use the wrong Client.
Each person can fail. but came to report problems There might be some points that you really missed.
And I've already reported the problem and it's really wrong.

This is the same problem.
Sorry translated by google.

Screenshot_sssss

**Comment by fab10Carvalho** [comment]
> Bug report: When join the party disconnect from server Error in Map Server
> 
> [Warning]: clif_parse: Received packet 0x0ae6 with expected packet length 12, but only 10 bytes remaining, disconnecting session #3.

Please Update your emulator!!

**Comment by gravity-ro** [comment]
> > Bug report: When join the party disconnect from server Error in Map Server
> > [Warning]: clif_parse: Received packet 0x0ae6 with expected packet length 12, but only 10 bytes remaining, disconnecting session #3.
> 
> Please Update your emulator!!  

 Thank you.
But i update from here everyday I don't know where I went wrong.

**Comment by ecdarreola** [comment]
Once you refuse and block a request it will always be permanent.

**Comment by zdurexz** [comment]
i will waiting update current.

**Comment by OptimusM** [comment]
Thanks @Lemongrass3110 for doing this amazing PR.

2 observations testing this:

1.-
I tried this feature on 2 clients, on clients from 2021, is everything working just fine, but, on 2022 clients, even march 2022, it has some response problem, i mean, everyone can register their own party, but, clicking on Join, does not generate any request to Party leader, and console says "[Info]: *ip* [POST/ party/info] 404
and the windows that should ask the party leader to accept or deny, does not show up. (to be clear, this is working ok on 2021 clients), so, seems like a packet problem maybe.

2.- 
if the map server has some problem, and somehow crash by any reason, the sql tables will keep the information of the registered party there, so you should manually clean the table to drop the information. (so, even if people delete the party, the information of the party, on the window will still show up.)


Thanks for the hard work!

**Comment by ecdarreola** [comment]
> > Once you refuse and block a request it will always be permanent.
> 
> @ecdarreola works normally for me. Make sure you dont tick that box: grafik

Yeah. I accidentally ticked the box. After that I can't seem to receive any invite from same account with different character. I'll take note of it.

**Comment by tmdwlse1990** [comment]
screenrAthena000
The Korean name of the Adventurer Agency is broken.
Do you have any solution?

**Comment by ecdarreola** [comment]
I experienced memory leak in partybooking_controller.cpp as attached.

image

**Comment by zdurexz** [comment]
Could you update this PR with the current version of rAthena?

**Comment by Lemongrass3110** [comment]
> Could you update this PR with the current version of rAthena?

@zdurexz done

**Comment by zdurexz** [comment]
> > Could you update this PR with the current version of rAthena?
> 
> @zdurexz done

Thanks @Lemongrass3110

**Comment by tmdwlse1990** [comment]
There is a phenomenon that Korean characters are broken when used in the euckr character set. Is there any solution?
Sorry for my weak English.
@Lemongrass3110
screenrAthena000

**Review by aleos89 — COMMENTED** [review]
image

**Pull request merged**